### PR TITLE
Log routing

### PR DESCRIPTION
**Pull request by meliaj**
Migrates sphinx to use log-routing and route error logs to SignalFX. Converts this rule: https://github.com/Clever/log-router/blob/0a195c6b4a7a339edb31e98ea16e1756ff4fb60b/config/apps.toml#L420

Forced an error in daemon ServeHTTP and it correctly sets `routes`:
```
{
  "_kvmeta":{
    "kv_language":"go",
    "kv_version":"6.6.0",
    "routes":[{
      "dimensions":["limit"],
      "rule":"sphinx-error",
      "series":"sphinx.error",
      "stat_type":"counter",
      "type":"alerts",
      "value_field":"value"
    }],
    "team":"UNSET"
  },
  "level":"error",
  "source":"sphinx",
  "title":"test"
}
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -9,13 +9,21 @@ import (
...
 )
 
 var (
 	configfile = flag.String("config", "example.yaml", "/path/to/configuration.yaml")
 	validate   = flag.Bool("validate", false, "Validate configuration and exit")
 )
 
+func init() {
+	if err := logger.SetGlobalRouting("kvconfig.yml"); err != nil {
+		log.Fatalf("error setting kv config: %s %s", "blah", err)
```
Comment (xavi-) [inline main.go]: probably should remove "blah"

**Comment by meliaj** [comment]
Done. I've built it and run it locally (with a forced error) to test the log routing happens

**Comment by meliaj** [comment]
Also in trying to get the forced error to happen, I saw that it does work otherwise :D

File: circle.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ test:
   - $HOME/ci-scripts/circleci/report-card $RC_DOCKER_USER $RC_DOCKER_PASS "$RC_DOCKER_EMAIL" $RC_GITHUB_TOKEN
 deployment:
   all:
-    branch: master
+    branch: log-routing
```
Comment (kvigen) [inline circle.yml]: reminder to switch this back

File: daemon/daemon.go [hunk 3]
```diff
@@ -84,7 +84,7 @@ func (d *daemon) LoadConfig(newConfig config.Config) error {
 		return fmt.Errorf("unrecognized handler %s", d.proxy.Handler)
 	}
 
-	d.handler = middleware.New(handler, common.Log, func(req *http.Request) map[string]interface{} {
+	d.handler = middleware.New(handler, "sphinx", func(req *http.Request) map[string]interface{} {
```
Comment (kvigen) [inline daemon/daemon.go]: should we get rid of the common.Log stuff now?

Comment (meliaj) [inline daemon/daemon.go]: /shrug

We're still using it elsewhere, so I'm leaving it for now

**Comment by meliaj** [comment]
Tested this end to end:
- building this branch
- changing sphinx-api to use this branch image and building that branch
- deploying the sphinx-api branch
- making a call to sphinx-api

👍

**Pull request merged**